packages/central-server/src/apiV2/utilities/constructNewRecordValidationRules.js

```diff
@@ -186,6 +186,74 @@ export const constructForSingle = (models, recordType) => {
         code: [hasContent],
         name: [hasContent],
       };
+    case TYPES.PROJECT:
+      return {
+        code: [
+          async projectCode => {
+            const projects = await models.project.find({
+              code: projectCode,
+            });
+            if (projects.length > 0) {
+              throw new Error('This project code already exists');
+            }
+            return true;
+          },
+        ],
+        name: [
```

Comment:
@IgorNadj I thought that perhaps given that there are a finite number of projects, and given that they all get assembled together for the user to view, it might be odd to have the same name.